e2etest/bgp_container.go

```diff
@@ -0,0 +1,189 @@
+		containerName).CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to get FRR IP address")
+	}
+
+	containerIPv4 := strings.TrimSuffix(string(containerIP), "\n")
+
+	containerIP, err = exec.Command("docker", "inspect", "-f", "{{range .NetworkSettings.Networks}}{{.GlobalIPv6Address}}{{end}}",
+		containerName).CombinedOutput()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to get FRR IP address")
+	}
+
+	containerIPv6 := strings.TrimSuffix(string(containerIP), "\n")
+
+	return containerIPv4, containerIPv6, nil
+}
+
+// Change volume permissions.
+// Allows deleting the test directory or updating the files in the volume.
+func updateFRRContainerVolumePermissions(containerName string) error {
+	out, err := exec.Command("id", "-u").CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get UID")
+	}
+	uid := strings.TrimSuffix(string(out), "\n")
+
+	out, err = exec.Command("id", "-g").CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get GID")
+	}
+	gid := strings.TrimSuffix(string(out), "\n")
+
+	cmd := fmt.Sprintf("chown -R %s:%s %s", uid, gid, FRRContainerMountDestination)
+	out, err = exec.Command("docker", "exec", containerName, "sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to change %s container volume permissions. %s", containerName, string(out))
+	}
+
+	return nil
+}
+
+// Delete the BGP router configuration.
+func stopFRRContainer(containerName string, testDirName string) error {
+	err := updateFRRContainerVolumePermissions(containerName)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to update %s container volume permissions", containerName)
+	}
+
+	// Kill the BGP router container.
+	out, err := exec.Command("docker", "kill", containerName).CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to kill %s container. %s", containerName, out)
+	}
+
+	err = os.RemoveAll(testDirName)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to delete FRR config directory.")
+	}
+
+	return nil
+}
+
+// Set the IP of each node in the cluster in the BGP router configuration.
+// Each node will peer with the BGP router.
+func bgpPeersForAllNodes(cs clientset.Interface) (string, error) {
+	nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to get cluster nodes")
+	}
+
+	// TODO: The IP address handling will need updates to add support for IPv6.
+	nodesIPs := make([]string, 0, len(nodes.Items))
+	for _, node := range nodes.Items {
+		for i := range node.Status.Addresses {
+			if node.Status.Addresses[i].Type == "InternalIP" {
+				nodesIPs = append(nodesIPs, node.Status.Addresses[i].Address)
+			}
+		}
+	}
+
+	if len(nodesIPs) < 3 {
+		return "", fmt.Errorf("Expected 3 nodes, got %d", len(nodesIPs))
+	}
+
+	bgpPeersInfo := "router bgp 64512\n"
```

Comment:
How about using a template here too (with a local template, no need to have an external file)?